services/repo_providers.py

```diff
@@ -70,7 +70,9 @@ def get_generic_adapter_params(owner: Owner, service, use_ssl=False, token=None)
             key=getattr(settings, f"{service.upper()}_CLIENT_ID", "unknown"),
             secret=getattr(settings, f"{service.upper()}_CLIENT_SECRET", "unknown"),
         ),
-        on_token_refresh=get_token_refresh_callback(owner, service),
+        on_token_refresh=(
+            get_token_refresh_callback(owner, service) if "username" in token else None
```

Comment:
What is `if "username" in token` checking for here?

Reply:
Oh I saw the code in https://github.com/codecov/worker/pull/69 about checking whether we're using the GH app integration.  Is this a proxy for that?  If so maybe just a quick comment here

Reply:
Yeah that's the idea. Since we define "username" in the token on line 62 - IFF we know what's the owner of said token - then we can use that to check if the token belongs to an owner (and thus can/should be refreshed).

I'll update with a comment